### ISSUE TITLE
ispc return code incorrect with #error preprocessor directive

### ISSUE THREAD

**Issue by kjeffery** [ispc/ispc#1212]
With this simple ispc file:

```
#error Error 1
#error Error 2

void f()
{
}
```

The return code of ispc reports success.

```
[kjeffery@xlimepony tmp]$ /rel/third_party/intelispc/1.9.0/ispc error.ispc -o error.o
Warning: No --target specified on command-line. Using default system target "avx1.1-i32x8". 
error.ispc:2:2: error: Error 1
#error Error 1
 ^
error.ispc:3:2: error: Error 2
#error Error 2
 ^
[kjeffery@xlimepony tmp]$ echo $?
0
```

**Comment by dbabokin** [comment]
`#error` seems to be handled by preprocessor, but it's not clear how this error state is communicated back to the preprocessor caller. Should be easy to fix, but requires some investigation.

**Comment by dbabokin** [comment]
It's basically same as #800

**Comment by dbabokin** [comment]
Fixed by #2280
